tokio/src/executor/task/harness.rs

```diff
@@ -51,7 +51,7 @@ where
     /// Panics raised while polling the future are handled.
     ///
     /// Returns `true` if the task needs to be scheduled again
-    pub(super) fn poll(mut self, executor: NonNull<S>) -> bool {
+    pub(super) fn poll(mut self, executor: &mut dyn FnMut() -> Option<NonNull<S>>) -> bool {
```

Comment:
Did you check how this impacts benchmarks? My preference is to use the `S: Scheduler` generic to avoid additional dynamic dispatch in the hot path.

Reply:
So, to be clear, this method is only actually invoked the first and last time a task is invoked. It is not called on every `poll`. This will matter for short-lived tasks, but not for long-lived ones. It could be that that's causing the smaller regressions, but it shouldn't incur any scalability overhead.